planner/core/preprocess.go

```diff
@@ -459,6 +460,30 @@ func (p *preprocessor) checkCreateViewGrammar(stmt *ast.CreateViewStmt) {
 	}
 }
 
+func (p *preprocessor) checkCreateViewWithSelect(stmt *ast.SelectStmt) {
+	if stmt.SelectIntoOpt != nil {
+		p.err = ddl.ErrViewSelectClause.GenWithStackByArgs("INFO")
+		return
+	}
+	if stmt.LockTp != ast.SelectLockNone {
+		stmt.LockTp = ast.SelectLockNone
+		return
+	}
+}
+
+func (p *preprocessor) checkCreateViewWithSelectGrammar(stmt *ast.CreateViewStmt) {
+	switch stmt := stmt.Select.(type) {
+	case *ast.SelectStmt:
+		p.checkCreateViewWithSelect(stmt)
+		break
+	case *ast.UnionStmt:
+		for _, selectStmt := range stmt.SelectList.Selects {
+			p.checkCreateViewWithSelect(selectStmt)
```

Comment:
Check `if p.err != nil` and `break` here?

Reply:
if p.err is not nil, i think logical should already break, so i do not need to check here.

Reply:
how about check the select in recursion;
```
somthing like:
func (p *preprocessor) checkCreateViewWithSelectGrammar(stmt *ast.CreateViewStmt) {
        checkCreateViewWithSelect(stmt.Select)
}
func (p *preprocessor) checkCreateViewWithSelect (stmt *ast.xxx) {
       switch stmt := stmt.Select.(type) {
	   case *ast.SelectStmt:
                 xxx
                 break
       case  *ast.UnionStmt:
                 checkCreateViewWithSelect(child...)
}

Reply:
Yes, my poor english.